backend/apis/parse-docket-pdf.utils.js

```diff
@@ -89,13 +89,80 @@ function parseCharges(lines, sections) {
 }
 
 function parseAccountSummary(lines, sections) {
-  const accounts = [];
+  let subsections = [];
+  let balances = [];
+  const sectionLastIndex = sections.filter(section => {
+    return section.name === "PROCEEDINGS" && section.lineNumber;
+  })[0].lineNumber;
+
+  function addAccounts(line, index) {
+    const words = line.split(/\s/);
+    const isSubsection =
+      words.length > 0 &&
+      words[0].toUpperCase() === words[0] &&
+      !words.includes("CASE") && // prevents CASE NUMBER lines from defining a subsection
+      words[1]; // prevents line new page diving line from defining a subsection;
+
+    if (isSubsection) {
+      subsections.push({ name: line, lineNumber: index });
+    }
+
+    balances = subsections
+      .map((subsection, arrayIndex) => {
+        const endIndex =
+          arrayIndex + 1 !== subsections.length
+            ? subsections[arrayIndex + 1].lineNumber
+            : sectionLastIndex;
+
+        let amountDue,
+          amountPaid,
+          costType,
+          collection,
+          originalAmountDue,
+          amendedAmountDue;
+
+        for (let i = subsection.lineNumber; i < endIndex; i++) {
+          const line = lines[i];
+          const words = line.split(/\s/);
+
+          if (line.includes("Amount Due")) {
+            if (line.includes("Original")) {
+              originalAmountDue = words[words.length - 1];
+            } else if (line.includes("Amended")) {
+              amendedAmountDue = words[words.length - 1];
+            }
+            amountDue = words[words.length - 1];
+          } else if (line.includes("Amount Paid") || line.includes("Paid In")) {
+            amountPaid = words[words.length - 1];
+          }
+          // console.log(line);
+          costType = getCostType(line) || costType;
+          collection = line.includes("State Debt Collection");
+        }
+        const returnObj = {
+          name: subsection.name,
+          amountDue,
+          originalAmountDue,
+          amendedAmountDue,
+          amountPaid,
+          costType,
+          collection
+        };
+        return amountDue && returnObj;
```

Comment:
There are cases where there are too many Amount Due lines. From https://drive.google.com/drive/u/0/folders/1cW6RM4jLDeiLeSy7emI8_xdf6HhpfBhG
we have this pair of lines: 

 Original Amount Due:         600.00
 Amended Amount Due:           0.00

The code was always returning 0.00 for the Amount Due because the second line trumped the first. That's why I added the two different properties so that both pieces of information would be preserved. Such cases always have an Amount Due part on a line, so I kept that property because it was the trigger to return the returnObj on Line 151. 

If we have good reason to believe that the use of Original Amount Due and Amended Amount due are only and always used in Debt Collection cases, then we could assign the amountDue property as in the original code (returning 0.00) and make use of the collection property when testing for qualification. If not, we would need to keep both in order to make the proper calculation during the qualification step. In the latter case, I suppose the Amended Amount Due would be the most likely candidate to return as the amountDue property, but I don't know about that.

Reply:
👍 yeah I saw the amendedAmountDue in the code and it seems to be a good thing to extract as a separate property.

My original question remains though - when will `amountDue` be undefined? The state debt collection cases seem to have both an `amountDue` and `amendedAmountDue`. So they don't seem like a situation where `amountDue` would be undefined.

Reply:
Amount Due will always be there so amountDue will never be undefined, as far as I can tell. If there's anything at all in the account summary, at least one line will have "Amount Due" on it. 

There's something that is bothering me about the code as written. We are currently getting this:
 {
      "name": "TRUST DETAIL",
      "amountDue": "0.00",
      "amountPaid": "0.00",
      "collection": false
    },

for anything that is a TRUST DETAIL. The actual raw file has this:

 TRUST DETAIL 
                 Trust Description: Interest on Rstitutn

Wouldn't we prefer to have the name = Interest on Rstitutn, or whatever comes to the right of the colon in the Trust Description line? What do you think? The eligibility calculator will need to know what the content of the TRUST DETAIL items are since the one I've printed here matters for eligibility but one like: 

 TRUST DETAIL 
                 Trust Description: Interest Attrny Fees
doesn't.

Reply:
```let trustName;
...
 if (line.includes("Trust Description")) {
            trustName = line.slice(line.indexOf(": ") + 2);
          }
...
const returnObj = {
...
trustName
}```

This code does the trick.

Reply:
I didn't read all of your comment because it's really long and not related to the question I asked. However, the first part of your comment is what I was asking about:

> Amount Due will always be there so amountDue will never be undefined, as far as I can tell

With that being the case, we should not have the `filter()` at all. And we should update this line of code, too, to not check amountDue for truthiness before returning it:

```suggestion
        return returnObj;
```

Reply:
Turns out that Bail is reported without an Amount Due: 
 BAIL/CASH BOND DETAIL - TYPE: BAIL                                     
                         Posted By: PERSON
                            Posted:      1,000.00
                         Forfeited:        781.65
                          Refunded:        218.35
                           Balance:          0.00
In the absence of the amountDue check, the above pushes to the accountSummary array like this:
 {
      "name": "BAIL/CASH BONDS    Posted:      1,000.00",
      "collection": false
    },
The accountSummary array also gets the Totals sections, which lack an Amount Due, like this:
 {
      "name": "ACCOUNT SUMMARY",
      "collection": false
    },
{
      "name": "TRUST TOTALS    Trust Due:       1,050.00",
      "amountPaid": "1,050.00",
      "collection": false
    },

I must have determined that those objects were noise and therefore left them out with the line 151 in the code.

What do you think?

Reply:
I think we should not leave out any of the subsections, even if the amount due is missing

Reply:
Done.